test/features/environment.py

```diff
@@ -68,12 +69,27 @@ def after_all(context):
     #        os.environ[k] = v
 
 
+def before_step(context, _):
+    context.atprompt = False
+
+
 def after_scenario(context, _):
     """Cleans up after each test complete."""
 
     if hasattr(context, 'cli') and not context.exit_sent:
-        # Terminate nicely.
-        context.cli.terminate()
+        # Quit nicely.
+        if not context.atprompt:
```

Comment:
I think the cleanup code makes sense in `after_scenario`, because we don't want to make the scenarios too boilerplate. In fact, I was thinking that the first 2 steps that are present in every scenario:

```
When we run dbcli
and we wait for prompt
```

could also be pulled out into the `before_scenario` handler. The scenarios are already becoming verbose, especially the CRUD ones that include multiple steps.